etc/dbus-serialbattery/bms/lltjbd_ble.py

```diff
@@ -268,6 +290,24 @@ def reset_bluetooth(self):
         sleep(5)
         sys.exit(1)
 
+    def reset_hci_uart(self):
+        logger.error("Reset of hci_uart stack... Reconnecting to: " + self.address)
+        self.run = False
+        os.system("pkill -f 'hciattach'")
+        sleep(0.5)
+        os.system("rmmod hci_uart")
+        os.system("rmmod btbcm")
+        os.system("modprobe hci_uart")
+        os.system("modprobe btbcm")
+        sys.exit(1)
```

Comment:
```
# start hciattach
/usr/bin/hciattach -t 10 /dev/ttyS0 bcm43xx 921600 noflow - &
```
https://github.com/Louisvdw/dbus-serialbattery/issues/777#issuecomment-1729239878 <- here you said that this line would be needed, but it's not part of the code changes. I was wondering if it is not needed after all?

Reply:
I don't find where I said that. Is the link maybe wrong? :-)

Reply:
The link leads to a comment by @Marvo2011 ;-)

Reply:
Ah sorry now I get it :-) @Marvo2011 any progress with this?

Reply:
Hello, I'm very busy at this weekend, I will take a look next week. For me it works but it's not perfect because it can take over a minute and in this time the system will shutdown the multi and other load outputs.

Reply:
Which part of your system controls if the inverter gets shut down when there is no connection between dbus-serialbattery and the BMS?
I would say it is perfectly fine to lose connection to the BMS for a few minutes, IF, let's say, the SOC is between 40% and 80%. What could go wrong in a few minutes until the connection is back? The Victron can still measure the total voltage of the battery, and the BMS is still functional itself (cell over/undervoltage protection)...

Reply:
@randomname32 this function does it https://github.com/Louisvdw/dbus-serialbattery/blob/c6d5bb6fcdc0d08404a0849345d6c332536b710c/etc/dbus-serialbattery/dbushelper.py#L597
We applied the default Victron Standard.

Reply:
@andomname32 I read the needed hciattach start command from ps and hold it down to the file: `/tmp/dbus-blebattery-hciattach` Line 58 execute it.

Reply:
> @randomname32 a lot can go wrong, since on many BMS the SOC can be completely wrong, but the approach is good. I would use the cell voltage instead. If the min cell voltage is above 3.2 and the max cell voltage below 3.3. What do you think?

Sounds good to me. But I am not an expert here, so maybe just add it as an optional thing...

Btw, today I found this option: 
```
; --------- BMS disconnect behaviour ---------
; False: Charge and discharge is not blocked on BMS communication loss
BLOCK_ON_DISCONNECT = False 
```
Is this not taken into consideration for BLE connections?

Reply:
sorry, you have pointed to that location before, but I did not fully read the code